### ISSUE TITLE
Setting up joints is difficult

### ISSUE THREAD

**Issue by briansemrau** [briansemrau/godot_box2d#11]
Setting up joints and their anchors is difficult with the editor. It would be nice to have editor handles, similar to what exists for shapes, to directly place anchors and modify joint properties.

**Comment by jordo** [comment]
have you used r.u.b.e at all?

**Comment by briansemrau** [comment]
I have not, but I imagine their solution is well thought out. And it might even suggest some changes to the not-thought-out-at-all solution I have to configuring custom joint anchors, if you've looked at how that's done right now.

**Comment by jordo** [comment]
:D I haven't looked yet, but will get there eventually.   I haven't used r.u.b.e in a while, but once you learn the hotkeys things like anchors on joints are one of the nice things to edit.  Stuff like this:  https://www.youtube.com/watch?v=YwHfhb3O_l

**Comment by jordo** [comment]
full playlist here:  https://www.youtube.com/playlist?list=PLzidsatoEzejt4smjfpeOd-XDjtE6PFr9   there may be an opportunity to take some inspiration from how the joint editor work there i dunno

**Comment by jordo** [comment]
I totally forgot about features like this, but at the end of this video https://www.youtube.com/watch?v=dTu2j9swqnY&list=PLzidsatoEzejt4smjfpeOd-XDjtE6PFr9&index=9 you can see that r.u.b.e shows a red line when anchor points are configured as disjoint, which is pretty cool.

**Comment by jordo** [comment]
I'm curious what the best work-flow for doing joints in Godot will be.  In general a tree isn't the best representation for a box2d world (you can string a bunch of joints and bodies together to make circle, think a tank tread).  Can you connect 6 bodies in a circle with joints in-between them?

**Comment by briansemrau** [comment]
So the only value of joints being in the tree is for positioning them. Right now their anchors are local to their transform. You do not need to add a joint as a child of anything other than the Box2DWorld.

If you want to make a bridge, it could be:
```
bridge (Node2D)
 L  Rect (body)
 L  Rect2 (body)
    ...
 L  RevoluteJoint (ground to 1)
 L  RevoluteJoint (1 to 2)
    ...
```

Or you could have
```
bridge (Node2D)
 L  Rect (body)
      L  RevoluteJoint (ground to 1)
 L  Rect2 (body)
      L  RevoluteJoint (1 to 2)
...
```

**Comment by briansemrau** [comment]
I took from how Godot joints work. Each joint has two nodepaths to connect bodies, connect absolutely any bodies in the tree

**Comment by jordo** [comment]
👍 👍

**Comment by briansemrau** [comment]
Should be generally resolved by #16. From now on, the joint editor plugin should be updated whenever a new joint is implemented, time permitting.
